Topics/Tech_Stacks/htmx.md

```diff
@@ -0,0 +1,83 @@
+and Server Sent Events. This simplicity and power of HTMX opens up a new world for web development.
+
+## Getting Started
+To start using HTMX, include its library in your project by adding a script tag to your HTML file. 
+This can be done in a couple different ways:
+
+
+effortlessly by referencing HTMX from a Content Delivery Network (CDN). 
+For example, to include HTMX, you could use the script below to get started (provided by the offical HTMX documentation):
+
+### Content Delivery Network (CDN)
+The first way to include HTMX is by referencing HTMX from a Content Delivery Network (CDN). This can be done by including the script below (Taken from the offical HTMX documentation):
+
+```html
+<script src="https://unpkg.com/htmx.org@1.9.11" integrity="sha384-0gxUXCCR8yv9FM2b+U3FDbsKthCI66oH5IA9fHppQq9DDMHuMauqq1ZHBpJxQ0J0" crossorigin="anonymous"></script>
+```
+
+### Copying HTMX into Directory
+You could also copy the HTMX file into your project directory. You can download the `htmx.min.js` file from [unpkg.com](https://unpkg.com/htmx.org@1.9.11/dist/htmx.min.js) then include it with this script tag:
+
+```html
+<script src="/path/to/htmx.min.js"></script>
+```
+
+### npm
+if you are using an npm based build system, you can use the following command to add it to your project:
+
+`npm install htmx.org`
+
+
+## AJAX Requests
+HTMX revolutionizes the way AJAX requests are handled directly from HTML, 
+eliminating the need for writing JavaScript code for these tasks. 
+You can easily make GET, POST, and other types of HTTP requests by adding specific HTMX attributes to your HTML elements.
+
+| Attribute | Description |
+|-----------------|-----------------|
+| hx-get | a `GET` request to the URL input |
+| hx-post | a `POST` request to the URL input |
+| hx-put | a `PUT` request to the URL input |
+| hx-patch | a `PATCH` request to the URL input |
+| hx-delete | a `DELETE` request to the URL input |
+
+
+Here is a simple example of using HTMX to make a `GET` request using the `hx-get` attribute:
+```html
+<button hx-get="/api/news" hx-trigger="click" hx-swap="outerHTML" hx-target="#news-article">
+    Load Cool News
+</button>
+<div id="news-article">Old not cool news content...</div>
+```
+This button, when clicked, loads the latest news into #news-section by completely replacing its contents.
+
+
+Here is another example of how to send data payloads using `POST` requests using the `hx-post` attribute:
+
+```html
+<form id="user-form" hx-post="/api/users/create" hx-target="#result">
+    <input type="text" name="username" placeholder="Username">
+    <input type="email" name="email" placeholder="Email">
+    <button type="submit">Create User</button>
+</form>
+<div id="result"></div>
+```
+When the form is submitted, HTMX sends a POST request to `/api/users/create` with the form data, and the server's response is displayed in `#result`.
+
```

Comment:
Maybe include a few more examples of more complicated requests (I.E. `UPDATE`, `PUT`, etc...), as well as a section discussing the limitations of HTMX.